sway/tree/view.c

```diff
@@ -785,7 +785,18 @@ void view_map(struct sway_view *view, struct wlr_surface *wlr_surface,
 
 	view_execute_criteria(view);
 
-	if (should_focus(view)) {
+	bool set_focus = should_focus(view);
+
+#if HAVE_XWAYLAND
+	if (wlr_surface_is_xwayland_surface(wlr_surface)) {
+		struct wlr_xwayland_surface *xsurface =
+			wlr_xwayland_surface_from_wlr_surface(wlr_surface);
+    	set_focus = (wlr_xwayland_icccm_input_state(xsurface) !=
```

Comment:
This overwrites the previous value of `set_focus`.

Do we want to always focus X11 windows that accept it? I don't think so, Sway can be configured to not give focus to new windows by default. So maybe we can add `&& set_focus` to the `if`?

Reply:
I've tested it both ways.  The fix only works if we override.